### PR TITLE
DEP remove old C/C++ examples

### LINKED ISSUE #714: DEP remove c/c++ examples
We should remove the old examples in c and c++.

### PR DESCRIPTION
**Pull request by beckermr**
Closes #714

### CODE REVIEW

**Comment by damonge** [comment]
But tests are failing

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -1,17 +0,0 @@
-add_executable(ccl_sample_baryons ccl_sample_baryons.c)
```
Comment (damonge) [inline examples/CMakeLists.txt]: I wonder if the problem has to do with having deleted this. Maybe the root CMakeLists.txt should be changed too.

Comment (beckermr) [reply examples/CMakeLists.txt]: I remove the ref to it from the main one. Should work now I think.

**Comment by beckermr** [comment]
OK @damonge this is working now.

**Pull request merged**